faker/providers/python/__init__.py

```diff
@@ -55,6 +55,40 @@ def _check_signature(self, value_types: Optional[TypesSpec], allowed_types: Opti
             allowed_types = ()
         return tuple(value_types) + tuple(allowed_types)
 
+    def pyobject(
+        self,
+        object_type: Optional[Type[Union[bool, str, float, int, tuple, set, list, Iterable, dict]]] = None,
+    ) -> Optional[Union[bool, str, float, int, tuple, set, list, Iterable, dict]]:
```

Comment:
I think we should be able to use some version of [`TypeVar`](https://docs.python.org/3/library/typing.html#typing.TypeVar) here:

```python3
TObjectType = TypeVar('TObjectType', Type[Union[None, bool, str, float, int, tuple, set, list, Iterable, dict]])

class Provider:
    ...
    def pyobject(self, object_type: TObjectType = None): -> TObjectType:
        ...
```

Reply:
Good suggestion.

Beware of the return type, which is not a type

Reply:
oh, that's right. I wonder if something like this would work?

```python3
TObjectType = TypeVar('TObjectType', Union[None, bool, str, float, int, tuple, set, list, Iterable, dict])

class Provider:
    ...
    def pyobject(self, object_type: Type[TObjectType] = None): -> TObjectType:
        ...
```

Reply:
It doesn't work with mypy